harmonica/_io/oasis_montaj_grd.py

```diff
@@ -0,0 +1,422 @@
+    # File offset from start of every block
+    block_offsets = array.array("q", grid_compressed[16 : 16 + n_blocks * 8])
+    # Compressed size of every block
+    compressed_block_sizes = array.array(
+        "i",
+        grid_compressed[16 + n_blocks * 8 : 16 + n_blocks * 8 + n_blocks * 4],
+    )
+    # Combine grid
+    grid = b""
+    # Read each block
+    for i in range(n_blocks):
+        # Define the start and end offsets for each compressed blocks
+        # We need to remove the 512 to account for the missing header.
+        # There is an unexplained 16 byte header that we also need to remove.
+        start_offset = block_offsets[i] - 512 + 16
+        end_offset = compressed_block_sizes[i] + block_offsets[i] - 512
+        # Decompress the block
+        grid_sub = zlib.decompress(
+            grid_compressed[start_offset:end_offset],
+            bufsize=zlib.DEF_BUF_SIZE,
+        )
+        # Add it to the running grid
+        grid += grid_sub
+    return grid
+
+
+def _build_coordinates(west, south, shape, spacing):
+    """
+    Create the coordinates for the grid
+
+    Generates 1d arrays for the easting and northing coordinates of the grid.
+    Assumes unrotated grids.
+
+    Parameters
+    ----------
+    west : float
+        Westernmost coordinate of the grid.
+    south : float
+        Southernmost coordinate of the grid.
+    shape : tuple
+        Tuple of ints containing the number of elements along each direction in
+        the following order: ``n_northing``, ``n_easting``
+    spacing : tuple
+        Tuple of floats containing the distance between adjacent grid elements
+        along each direction in the following order: ``s_northing``,
+        ``s_easting``.
+
+    Returns
+    -------
+    easting : 1d-array
+        Array containing the values of the easting coordinates of the grid.
+    northing : 1d-array
+        Array containing the values of the northing coordinates of the grid.
+    """
+    easting = np.linspace(west, west + spacing[1] * (shape[1] - 1), shape[1])
+    northing = np.linspace(south, south + spacing[0] * (shape[0] - 1), shape[0])
+    return easting, northing
+
+
+def _build_rotated_coordinates(west, south, shape, spacing, rotation_deg):
+    """
+    Create the coordinates for a rotated grid
+
+    Generates 2d arrays for the easting and northing coordinates of the grid.
+    Assumes rotated grids.
+
+    Parameters
+    ----------
+    west : float
+        Westernmost coordinate of the grid.
+    south : float
+        Southernmost coordinate of the grid.
+    shape : tuple
+        Tuple of ints containing the number of elements along each unrotated
+        direction in the following order: ``n_y``, ``n_x``
+    spacing : tuple
+        Tuple of floats containing the distance between adjacent grid elements
+        along each unrotated direction in the following order: ``spacing_y``,
+        ``spacing_x``.
+
```

Comment:
add docstrings for `rotation_deg`? 

```
rotation_deg : float
     Grid axis rotation, in degrees counter clockwise, relative to the co-ordinate system axis.
```

Reply:
Nice catch! Thanks for including it.

BTW, you can make suggestions during the review using the little icon that seems like a document with a + and - signs inside it. It generates a nice "Apply changes" button that creates the commit automatically, and it also gives you attribution to the PR by adding you as a co-author. No big deal, but just wanted to mention it because it's a cool feature to use.